modules/ckanext-ytp_main/ckanext/ytp/templates/package/snippets/dataset_state_banner.html

```diff
@@ -0,0 +1,37 @@
+{% if pkg.private %}
+  <div class="well admin-banner private-or-draft-dataset">
+    <div>
+      <img src="/resources/images/avoindata-closed-eye.svg">
+      <span>{{ _('Private mode. The dataset is only visible to logged in users of the producer organization.') }}</span>
+    </div>
+    {% if h.check_access('package_update', {'id':pkg.id }) %}
+      {% link_for _('Edit dataset'), controller='package', action='edit', id=pkg.name %}"
+    {% endif %}
+  </div>
+{% elif pkg.get('state', '').startswith('deleted') %}
+  <div class="well admin-banner private-or-draft-dataset deleted-dataset">
+    <div>
+      <img src="/resources/images/avoindata-closed-eye.svg">
+      <span>{{ _('Deleted. The dataset is only visible to logged in users of the producer organization.') }}</span>
+    </div>
+    {% if h.check_access('package_update', {'id':pkg.id }) %}
+      {% link_for _('Edit dataset'), controller='package', action='edit', id=pkg.name %}"
+    {% endif %}
+  </div>
+{% elif pkg.state.startswith('draft') %}
+  <div class="well admin-banner private-or-draft-dataset">
+    <div>
+      <img src="/resources/images/avoindata-closed-eye.svg">
+      <span>{{ _('In draft state. The dataset is only visible to logged in users of the producer organization.') }}</span>
+    </div>
+    {% if h.check_access('package_update', {'id':pkg.id }) %}
+      {% link_for _('Edit dataset'), controller='package', action='edit', id=pkg.name %}"
+    {% endif %}
+  </div>
+{% else %}
+  <div class="admin-banner">
+    {% if h.check_access('package_update', {'id':pkg.id }) %}
+      {% link_for _('Edit dataset'), controller='package', action='edit', id=pkg.name %}
+    {% endif %}
+  </div>
+{% endif %}
```

Comment:
This works, but one question pops up on my mind: Was there any plan in design for private dataset that is deleted, as now it will show just a private "flag" for all cases if dataset is private

Reply:
No plan for this in designs, I'll clarify with the designer.

Reply:
Same applied to the draft & private state. Now both cases show either the deleted or draft indicator even if they are also private.

Reply:
My previous comment seems to have disappeared 🤔 

The same situation happened with datasets that were both private & and in a draft state. Confirmed both of these cases with the designer and now draft & deleted states take precedence over private state.